### PR TITLE
[BREAKING] Named blocks

### PR DESCRIPTION
**Pull request by wycats**
Open pull request for implementing named blocks.

Breaking changes (will require updates in Glimmer.js and/or Ember):

- `invokeStaticComponent` API change

### CODE REVIEW

File: packages/@glimmer/opcode-compiler/lib/opcode-builder.ts [hunk 1]
```diff
@@ -747,22 +750,29 @@ export abstract class OpcodeBuilder<Locator = Opaque> extends StdOpcodeBuilder {
...
-    capabilities: ComponentCapabilities,
-    layout: CompilableProgram,
-    attrs: Option<CompilableBlock>,
-    params: Option<WireFormat.Core.Params>,
-    hash: WireFormat.Core.Hash,
-    synthetic: boolean,
-    block: Option<CompilableBlock>,
-    inverse: Option<CompilableBlock> = null
-  ) {
+  invokeStaticComponent({
```
Comment (rwjblue) [inline packages/@glimmer/opcode-compiler/lib/opcode-builder.ts]: I believe this makes it a breaking change, we should label the PR (and probably start a list of things for tracking in Ember update)...

File: package.json [hunk 2]
```diff
@@ -27,15 +28,16 @@
     "yarn:unlink": "node bin/yarn-link-all --unlink"
   },
   "ember-addon": {
-    "paths": [
-      "build/symlink-dependencies"
-    ]
+    "paths": ["build/symlink-dependencies"]
+  },
+  "resolutions": {
+    "typescript": "^3.1.0"
```
[the remark this answers is not in the document]
Comment (wycats) [reply package.json]: +1

File: .vscode/settings.json [hunk 3]
```diff
@@ -24,5 +24,6 @@
   "files.trimTrailingWhitespace": true,
   "editor.renderWhitespace": "boundary",
   "editor.insertSpaces": true,
-  "editor.tabSize": 2
+  "editor.tabSize": 2,
```
[the remark this answers is not in the document]
Comment (wycats) [reply .vscode/settings.json]: And this setting should come from editorconfig anyway, right?

File: packages/@glimmer/opcode-compiler/lib/opcode-builder.ts (hunk 1 above)
Comment (wycats) [reply packages/@glimmer/opcode-compiler/lib/opcode-builder.ts]: Yep!

File: .vscode/settings.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (wycats) [reply .vscode/settings.json]: Actually I'm not quite sure why you think this shouldn't be checked in. What about this file is user-specific?

[the remark this answers is not in the document]
Comment (btecu) [reply .vscode/settings.json]: The file is editor specific, so someone not using VS Code wouldn't have any use for it.

[the remark this answers is not in the document]
Comment (rwjblue) [reply .vscode/settings.json]: Right. We don't check in vim / emacs / sublime / atom / textmate / etc configs either (nor do we want them).

File: packages/@glimmer/opcode-compiler/lib/opcode-builder.ts (hunk 1 above)
Comment (rwjblue) [reply packages/@glimmer/opcode-compiler/lib/opcode-builder.ts]: OK, I added `BREAKING` to the PR title, and started a bulleted list for the things that we know are breaking. At some point, we should review the diff and fill in more known scenarios...

**Pull request merged**